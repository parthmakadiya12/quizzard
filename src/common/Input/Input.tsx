import React, { FC } from "react";
import styled from "styled-components";

export interface InputProps {
  type: string;
  value?: string | number;
  placeholder: string;
  changeHandler: (e: any, id?: string | number) => void;
  id?: string | number;
}

const Input: FC<InputProps> = ({
  type = "text",
  value,
  placeholder,
  changeHandler,
  id,
}) => {
  return (
    <InputStyled
      type={type}
      placeholder={placeholder}
      value={value || ""}
      onChange={(e) => changeHandler(e.target.value, id)}
    />
  );
};

const InputStyled = styled.input`
  height: 30px;
  outline: none;
  padding: 0 10px;
  border: ${({ theme }) =>
    theme.type === "dark" ? "none" : "1px solid black"};
  background-color: ${({ theme }) => theme.color.inputColor};
  color: ${({ theme }) => theme.color.inputFontColor};
  &::placeholder {
    color: ${({ theme }) => theme.color.inputFontColor};
    opacity: 0.6;
  }
`;

export default Input;
