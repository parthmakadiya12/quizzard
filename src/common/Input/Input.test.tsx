import Input, { InputProps } from "./Input";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import "jest-styled-components";

describe("<Input />", () => {
  let props: InputProps, theme: object, wrapper: any;
  beforeEach(() => {
    theme = {
      color: {
        inputColor: "#4a5171",
        inputFontColor: "white",
      },
      type: "dark",
    };
    props = {
      type: "text",
      value: "Current Value",
      placeholder: "Please choose",
      changeHandler: jest.fn(),
      id: "id",
    };
    wrapper = mount(
      <ThemeProvider theme={theme}>
        <Input {...props} />
      </ThemeProvider>
    );
  });
  it("should have rendered input correctly", () => {
    const input = wrapper.find("input");
    expect(input).toBeTruthy();
  });

  it("should have call click handler by clicking on button", () => {
    const button = wrapper.find("input");
    const event = {
      preventDefault() {},
      target: { value: "option A" },
    };
    button.simulate("change", event);
    expect(props.changeHandler).toHaveBeenCalledWith("option A", "id");
  });

  it("should render correctly with darkmode", () => {
    const newProps = { ...props, type: undefined };
    const newTheme = { ...theme, type: "light" };
    wrapper = mount(
      <ThemeProvider theme={newTheme}>
        <Input {...newProps} />
      </ThemeProvider>
    ).debug();

    expect(wrapper).toMatchSnapshot();
  });
});
