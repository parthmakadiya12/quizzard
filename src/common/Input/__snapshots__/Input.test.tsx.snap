// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render correctly with darkmode 1`] = `
"<Component theme={{...}}>
  <Input type={[undefined]} value=\\"Current Value\\" placeholder=\\"Please choose\\" changeHandler={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} id=\\"id\\">
    <styled.input type=\\"text\\" placeholder=\\"Please choose\\" value=\\"Current Value\\" onChange={[Function: onChange]}>
      <input type=\\"text\\" placeholder=\\"Please choose\\" value=\\"Current Value\\" onChange={[Function: onChange]} className=\\"sc-bdnxRM LhQCc\\" />
    </styled.input>
  </Input>
</Component>"
`;
